library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -47,103 +48,103 @@ public void migrate() {
 
         migrationCallback.onUpdate("Migrating Files");
         Log.d(TAG, "about to migrate the files, time is " + System.nanoTime());
-        migrateV1FilesToV2Location(migrations);
-        Log.d(TAG, "files are all MOVED, about to start migrating the database, time is " + System.nanoTime());
 
-        migrateV1DataToV2Database(migrations);
+        for (Migration migration : migrations) {
+            downloadsPersistence.startTransaction();
+            database.startTransaction();
+
+            migrateV1FilesToV2Location(migration);
+            migrateV1DataToV2Database(migration);
+            deleteFrom(database, migration);
+
+            downloadsPersistence.transactionSuccess();
+            downloadsPersistence.endTransaction();
+            database.setTransactionSuccessful();
+            database.endTransaction();
```

Comment:
End both transactions at the same time just in case the process is terminated during method execution.